experimental/dds/tree2/src/shared-tree-core/editManagerCodecs.ts

```diff
@@ -37,7 +41,7 @@ export function makeEditManagerCodec<TChangeset>(
 				]),
 			};
 			assert(format.check(json), 0x6cc /* Encoded schema should validate */);
-			return json;
+			return json as unknown as JsonCompatibleReadOnly;
```

Comment:
if you declare `EncodedCommit` using `type` and not `interface`, it will be given an implicit index signature that allows it to be assignable to `JsonCompatibleReadOnly`. That's probably what he's referring to.